ray-curator/ray_curator/stages/audio/inference/asr_nemo.py

```diff
@@ -0,0 +1,145 @@
+            Tuple of (top_level_attrs, data_attrs) where:
+            - top_level_attrs: ["data"] - requires FileGroupTask.data to be populated
+        """
+        return ["data"], []
+
+    def outputs(self) -> tuple[list[str], list[str]]:
+        """Define the output attributes produced by this stage.
+
+        Returns:
+            Tuple of (top_level_attrs, data_attrs) where:
+            - top_level_attrs: ["data"] - populates FileGroupTask.data
+            - data_attrs: [self.filepath_key, self.pred_text_key] - audiofile path and predicted text.
+        """
+        return ["data"], [self.filepath_key, self.pred_text_key]
+
+    def transcribe(self, files: list[str]) -> list[str]:
+        """Run inference for speech recognition model
+         Args:
+            files: list of audio file paths.
+
+        Returns:
+            list of predicted texts.
+        """
+        outputs = self.asr_model.transcribe(files)
+        return [output.text for output in outputs]
+
+    def process_batch(self, tasks: list[FileGroupTask | DocumentBatch | SpeechObject]) -> list[SpeechObject]:
+        """Process a audio task by reading audio file and do ASR inference.
+
+
+        Args:
+            tasks: List of FileGroupTask containing a path to audop file for inference.
+
+        Returns:
+            List of SpeechObject with self.filepath_key .
+            If errors occur, the task is returned with error information stored.
+        """
+
+        if isinstance(tasks[0], FileGroupTask):
+            files = [task.data[0] for task in tasks]
+        elif isinstance(tasks[0], DocumentBatch):
+            files = []
+            for task in tasks:
+                files.extend(list(task.data[self.filepath_key]))
+        elif isinstance(tasks[0], SpeechObject):
+            files = [task.data[self.filepath_key] for task in tasks]
+        else:
+            raise TypeError(str(tasks[0]))
+
+        outputs = self.transcribe(files)
+
+        audio_tasks = []
+        for i in range(len(outputs)):
+            entry = tasks[i].data
+            text = outputs[i]
+            file_path = files[i]
+
+            if isinstance(entry, dict):
+                item = entry
+                item[self.pred_text_key] = text
+            else:
+                item = {self.filepath_key: file_path, self.pred_text_key: text}
+
+            audio_task = SpeechObject(
+                task_id=f"task_id_{file_path}",
+                dataset_name=f"{self.model_name}_inference",
+                filepath_key=self.filepath_key,
+                data=item,
+            )
+            audio_tasks.append(audio_task)
+        self.finalize()
+        return audio_tasks
+
+    def process(self, task: FileGroupTask) -> list[SpeechObject]:
+        pass
+
+    def finalize(self) -> None:
+        elapsed = time.time() - self._start_time
+        logger.info(f"Stage {self.name} completed in {elapsed:.2f} seconds.")
+        self._metrics["elapsed"] = elapsed
```

Comment:
Don't think we need this. Can be consolidated with `_stage_perf` instead.

Reply:
removed